tests/integration/local/start_lambda/test_start_lambda_terraform_applications.py

```diff
@@ -0,0 +1,144 @@
+
+        if cls.terraform_application:
+            cls.working_dir = cls.integration_dir + cls.terraform_application
+
+        cls.port = str(StartLambdaIntegBaseClass.random_port())
+        cls.env_var_path = cls.integration_dir + "/testdata/invoke/vars.json"
+
+        if cls.build_before_invoke:
+            cls.build()
+
+        # remove all containers if there
+        cls.docker_client = docker.from_env()
+        for container in cls.docker_client.api.containers():
+            try:
+                cls.docker_client.api.remove_container(container, force=True)
+            except APIError as ex:
+                LOG.error("Failed to remove container %s", container, exc_info=ex)
+
+        cls.start_lambda()
+
+
+class TestLocalStartLambdaTerraformApplicationWithoutBuild(StartLambdaTerraformApplicationIntegBase):
+    terraform_application = "/testdata/invoke/terraform/simple_application_no_building_logic"
+    template_path = None
+    hook_package_id = "terraform"
+
+    def setUp(self):
+        self.url = "http://127.0.0.1:{}".format(self.port)
+        self.lambda_client = boto3.client(
+            "lambda",
+            endpoint_url=self.url,
+            region_name="us-east-1",
+            use_ssl=False,
+            verify=False,
+            config=Config(signature_version=UNSIGNED, read_timeout=120, retries={"max_attempts": 0}),
+        )
+
+    functions = [
+        "s3_lambda_function",
+        "aws_lambda_function.s3_lambda",
+        "root_lambda",
+        "aws_lambda_function.root_lambda",
+        "level1_lambda_function",
+        "module.level1_lambda.aws_lambda_function.this",
+        "level2_lambda_function",
+        "module.level1_lambda.module.level2_lambda.aws_lambda_function.this",
+    ]
+
+    @skipIf(
+        not CI_OVERRIDE,
+        "Skip Terraform test cases unless running in CI",
+    )
+    @parameterized.expand(functions)
+    @pytest.mark.flaky(reruns=3)
+    def test_invoke_function(self, function_name):
+        response = self.lambda_client.invoke(FunctionName=function_name)
+
+        response_body = json.loads(response.get("Payload").read().decode("utf-8"))
+        expected_response = json.loads('{"statusCode":200,"body":"{\\"message\\": \\"hello world\\"}"}')
+
+        self.assertEqual(response_body, expected_response)
+        self.assertEqual(response.get("StatusCode"), 200)
+
+
+class TestLocalStartLambdaInvalidUsecasesTerraform(TestCase):
+    def setUp(self):
+        self.integration_dir = str(Path(__file__).resolve().parents[2])
+        terraform_application = "/testdata/invoke/terraform/simple_application_no_building_logic"
+        self.terraform_application_path = self.integration_dir + terraform_application
+
+    def test_invalid_hook_package_id(self):
+        command = "sam"
+        if os.getenv("SAM_CLI_DEV"):
+            command = "samdev"
+
+        command_list = [command, "local", "start-lambda", "--hook-package-id", "tf"]
```

Comment:
This usually comes from `StartLambdaIntegBaseClass`, any reason for not having this in the common place?

Reply:
in StartLambdaIntegBaseClass, we do not send parameters to customize the local command

Reply:
also, in the base class .. it assumes that the start lambda command will be successful, and does not return the out or error stream. That is why, for the invalid use cases, I created a new class, and extended TestCase instead of extending the StartLambdaIntegBaseClass.

Reply:
Done